src/components/atom/AppDigitsValidator.tsx

```diff
@@ -0,0 +1,84 @@
+import React, { useRef } from "react";
+import DigitsValidatorProps from "../../types/DigitsValidatorProps";
+export const AppDigitsValidator = ({
+                    label,
+                    digitsCount,
+                    value,
+                    onChange,
+                    hasError,
+                    error,
+                    }: DigitsValidatorProps) => {
+    
+    const digitsRef = useRef<HTMLInputElement[]>([]);
+
+
+    const onDigitsChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+        const {value: fieldValue} = e.target;
+        const isOnlyNumbers = /^\d+$/.test(fieldValue);   
+
+        let insertedValue = (e.nativeEvent as InputEvent).data || '';
```

Comment:
```suggestion
         const insertedValue = (e.nativeEvent as InputEvent).data || '';
```

I would try to avoid `let` as much as possible. I believe that you can use `const` here instead